client/state/data-layer/wpcom/concierge/shifts/index.js

```diff
@@ -0,0 +1,50 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateConciergeShifts } from 'state/concierge/actions';
+import { errorNotice } from 'state/notices/actions';
+import { CONCIERGE_SHIFTS_REQUEST } from 'state/action-types';
+
+export const fetchConciergeShifts = ( { dispatch }, action ) => {
+	const { scheduleId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/concierge/schedules/${ scheduleId }/shifts`,
+				apiNamespace: 'wpcom/v2',
+			},
+			action
+		)
+	);
+};
+
+export const storeFetchedConciergeShifts = ( { dispatch }, action, shifts ) =>
+	dispatch( updateConciergeShifts( shifts ) );
+
+export const conciergeShiftsFetchErrorNotice = errorNotice(
+	translate( "We've encountered problems trying to load available shifts. Please try again later." )
+);
```

Comment:
Sadly this is also going against the grain of how translations work. Honestly I think it would probably be fine, but what it would end up in fact is a string that is only translated once at boot time and never updated. We have options around it, the simplest being making it a function…

```js
export const fetchError = () => errorNotice( translate( … ) );
```

now when we call it we run it and the translation will be produced on the spot in whatever the current language is.

Reply:
OOps. That's quite vital I believe. In the end, I decided to go to the old way with a clearer explanation in 6168064, since I think it's a better compromise to work around the auto-incremented underlying `noticeId`. Now I can finally understand why [countries list](https://github.com/Automattic/wp-calypso/blob/ff03397bbca59c5d56cfbc494123d9ddc7fa7a7d/client/state/data-layer/wpcom/domains/countries-list/test/index.js#L65) is doing this ...